apis/networking/v1beta1/apiproduct_types.go

```diff
@@ -87,6 +89,30 @@ func (a *APISelector) APINamespacedName() types.NamespacedName {
 	return types.NamespacedName{Namespace: a.Namespace, Name: name}
 }
 
+type AuthRateLimitSpec struct {
+	// MaxValue represents the number of requests allowed per defined period of time.
+	MaxValue int32 `json:"maxValue"`
+	// Period represents the period of time in seconds.
+	Period int32 `json:"period"`
+}
+
+type PreAuthRateLimitSpec struct {
+	// MaxValue represents the number of requests allowed per defined period of time.
+	MaxValue int32 `json:"maxValue"`
+	// Period represents the period of time in seconds.
+	Period int32 `json:"period"`
+}
+
+type RateLimitSpec struct {
+	// PreAuthRateLimitSpec configures rate limit parameters per remote address
+	// +optional
+	PreAuthRateLimit *PreAuthRateLimitSpec `json:"preAuthenticated,omitempty"`
```

Comment:
Porting the [discussion](https://github.com/Kuadrant/kuadrant-controller/pull/47#discussion_r705106910) here. 

Currently it looks like:
```yaml
spec:
  rateLimit:
    authenticated:
      maxValue: 5
      period: 30
    preAuthenticated:
      maxValue: 10
      period: 30
```

What do you propose @guicassolato @jmprusi ??

Reply:
```yaml
spec:
  rateLimit:
    network:
      maxValue: 10
      period: 40
    authenticated:
      maxValue: 5
      period: 30
```

Reply:
It can also be

```yaml
spec:
  rateLimit:
    pre-remote-ip:
      maxValue: 10
      period: 40
    authenticated:
      maxValue: 5
      period: 30
```

Then we can add global rate limit:

```yaml
spec:
  rateLimit:
   global:
      maxValue: 10
      period: 40
    per-remote-ip:
      maxValue: 10
      period: 40
    authenticated:
      maxValue: 5
      period: 30
```

Reply:
I ended up implementing

```yaml
spec:
  rateLimit:
    global:
      maxValue: 30
      period: 40
    perRemoteIP:
      maxValue: 10
      period: 40
    authenticated:
      maxValue: 5
      period: 40
```

which seems more flexible struct. Which can always be changed, of course.